engine/src/main/java/org/hibernate/validator/internal/engine/ValidationContext.java

```diff
@@ -196,6 +214,13 @@ public T getRootBean() {
 		return rootBeanClass;
 	}
 
+	public BeanMetaData<T> getRootBeanMetaData() {
+		if ( rootBeanClass != null && rootBeanMetaData == null ) {
```

Comment:
So in this case, it's an error. It's just that I started with `ValueContext` and then copied the behavior from `ValueContext` to `ValidationContext`. I'll change it here.